src/bindings/python/flux/job/Jobspec.py

```diff
@@ -741,6 +741,61 @@ def _v1_validate(resources, tasks, kwargs):
             if task["count"].keys().isdisjoint({"per_slot", "total"}):
                 raise ValueError("count per_slot or total must be set")
 
+    def _set_extra_options(
+        self,
+        duration=None,
+        environment=None,
+        env_expand=None,
+        cwd=None,
+        rlimits=None,
+        name=None,
+        stdin=None,
+        stdout=None,
+        stderr=None,
+        label_io=False,
+        unbuffered=False,
+        queue=None,
+        bank=None,
+    ):
+        """Process common jobspec factory args and return amended jobspec"""
+        if duration is not None:
+            self.duration = duration
+        if environment is not None:
+            self.environment = environment
+        if env_expand is not None:
+            if not isinstance(env_expand, abc.Mapping):
+                raise ValueError("env_expand must be a mapping")
+            self.setattr_shell_option("env-expand", env_expand)
+        if cwd is not None:
+            self.cwd = cwd
+        if rlimits is not None:
+            if not isinstance(rlimits, abc.Mapping):
+                raise ValueError("rlimits must be a mapping")
+            self.setattr_shell_option("rlimit", rlimits)
+        if name is not None:
+            self.setattr("job.name", name)
+        if stdin is not None:
+            self.stdin = stdin
+        if stdout is not None and stdout not in ("none", "kvs"):
+            self.stdout = stdout
+            if label_io:
+                self.setattr_shell_option("output.stdout.label", True)
+        if stderr is not None and stderr not in ("none", "kvs"):
+            self.stderr = stderr
+            if label_io:
+                self.setattr_shell_option("output.stderr.label", True)
+        if unbuffered:
+            #  For output, set the buffer.type to none and reduce the configured
+            #  event batch-timeout to something very small.
+            self.setattr_shell_option("output.stdout.buffer.type", "none")
+            self.setattr_shell_option("output.stderr.buffer.type", "none")
+            self.setattr_shell_option("output.batch-timeout", 0.05)
```

Comment:
This one seems to be the odd one out, in that it can be set with a keyword argument but not with a property is that right?  I think all of the others are both at this point.

Reply:
Yeah, we never did add a setter/getter for that one. Do you feel we should add that?

Oh, the other thing I did change what that the factory function arguments for file output are `stdout`, `stderr`, `stdin`, whereas the command line arguments are `output`, `error`, and `input`. I wonder if I should tweak them to match.

Reply:
I was waffling on it when I asked, but having given it some thought I think yes.  That way both modes work, and we could handle all they keyword args somewhat uniformly with a version of `if <name> is not None: self.name = name` since the logic can live in the property.  It's not a strong preference, but feels more regular to me somehow.

On the args, that's a good point.  The difference isn't ideal, I'm thinking maybe we add aliases that are the command line names, then if we want to squash the `std*` ones we could mark them deprecated for a while at some point.